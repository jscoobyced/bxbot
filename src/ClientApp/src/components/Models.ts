export class Constants {
    public static readonly StaticContent = (process.env.mode === 'development') ? '' : '/dist';

    public static readonly ImagePath = `${Constants.StaticContent}/images/`;
}

export interface SelectOption {
    value: number;
    text: string;
}
