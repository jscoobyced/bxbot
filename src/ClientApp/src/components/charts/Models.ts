export interface Pairing {
    timestamp: number;
    low: number;
    high: number;
    close: number;
    open: number;
}

export interface CandleChartPageState {
    pairings: Pairing[];
    loading: boolean;
}

export interface CandleChartProps {
    currency: string;
    element: string;
}