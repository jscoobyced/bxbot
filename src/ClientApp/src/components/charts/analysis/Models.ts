import { Pairing, Pattern } from "../Models";

export interface IAnalyser {
    analyse(pairings: Pairing[], index: number): Pattern;
}