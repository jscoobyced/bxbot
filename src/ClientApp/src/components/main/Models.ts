import { NavigationMenuContent } from "../../services/Models";

export interface HeaderProps {
    navigationMenuContent: NavigationMenuContent;
}
