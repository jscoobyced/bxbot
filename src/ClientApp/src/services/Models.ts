export enum Language {
    English,
    French
}

export interface AllContent {
    navigationMenuContent: NavigationMenuContent;
    homeContent: HomeContent;
}

export interface NavigationMenuContent {
    home: string;
    websiteName: string;
    cryptoCurrency: string;
    candleChart: string;
    about: string;
}

export interface HomeContent {
    articleTitle: string;
    article: string;
}