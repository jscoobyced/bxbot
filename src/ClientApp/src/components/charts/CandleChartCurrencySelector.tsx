import * as React from 'react';
import { CandleChartCurrencyProps } from './Models';

export class CandleChartCurrencySelector extends React.Component<CandleChartCurrencyProps> {

    public constructor(props: CandleChartCurrencyProps) {
        super(props);
    }

    public render() {
        return <select id='currencySelector'
                        onChange={this.props.onChangeCurrency}>
            {
                this.props.currencyOptions && this.props.currencyOptions.map((selectOption, key) => {
                    return <option key={key} value={selectOption.value}>{selectOption.text}</option>;
                })
            }
        </select>;
    }
}
