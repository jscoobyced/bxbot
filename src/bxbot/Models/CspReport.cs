using Newtonsoft.Json;

namespace bxbot
{
    public class CspReport
    {
        [JsonProperty(PropertyName = "document-uri")]
        public string DocumentUri { get; set; }

        [JsonProperty(PropertyName = "referrer")]
        public string Referrer { get; set; }

        [JsonProperty(PropertyName = "violated-directive")]
        public string ViolatedDirective { get; set; }

        [JsonProperty(PropertyName = "effective-directive")]
        public string EffectiveDirective { get; set; }

        [JsonProperty(PropertyName = "original-policy")]
        public string OriginalPolicy { get; set; }

        [JsonProperty(PropertyName = "blocked-uri")]
        public string BlockedUri { get; set; }

        [JsonProperty(PropertyName = "status-code")]
        public int StatusCode { get; set; }
    }
}