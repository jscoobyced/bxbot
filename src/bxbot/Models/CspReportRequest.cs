using Newtonsoft.Json;

namespace bxbot
{
    public class CspReportRequest
    {
        [JsonProperty(PropertyName = "csp-report")]
        public CspReport CspReport { get; set; }
    }
}