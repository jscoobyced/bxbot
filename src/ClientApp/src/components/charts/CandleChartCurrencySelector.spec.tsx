import { shallow } from 'enzyme';
import * as React from 'react';
import { CandleChartCurrencySelector } from './CandleChartCurrencySelector';
import { SelectOption } from '../Models';

test('CandleChartCurrencySelector snapshot', () => {
    const data: SelectOption[][] = [
        [],
        [{
            value: 1,
            text: 'THB/BTC'
        }]
    ];

    data.map(expected => {
        const candleChartCurrencySelector = shallow(<CandleChartCurrencySelector
            currencyOptions={expected}
            onChangeCurrency={() => { return; }} />);
        expect(candleChartCurrencySelector).toMatchSnapshot();
    })

});

test('CandleChartCurrencySelector onChangeCurrency', () => {
    const onChangeCurrency = jest.fn().mockImplementation(() => {
        return;
    });
    const candleChartCurrencySelector = shallow(<CandleChartCurrencySelector
        currencyOptions={[]}
        onChangeCurrency={onChangeCurrency} />);
    candleChartCurrencySelector
        .find('#currencySelector')
        .simulate('change', { target: { value: '25' }, preventDefault: () => { return; } });
    expect(onChangeCurrency).toHaveBeenCalledTimes(1);

});