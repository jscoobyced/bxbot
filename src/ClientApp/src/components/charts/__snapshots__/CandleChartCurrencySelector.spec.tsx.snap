// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CandleChartCurrencySelector snapshot 1`] = `
<select
  id="currencySelector"
  onChange={[Function]}
/>
`;

exports[`CandleChartCurrencySelector snapshot 2`] = `
<select
  id="currencySelector"
  onChange={[Function]}
>
  <option
    key="0"
    value={1}
  >
    THB/BTC
  </option>
</select>
`;
